src/hooks/useJournalData.ts

```diff
@@ -0,0 +1,252 @@
+/* eslint-disable no-await-in-loop */
+import { Auth } from '@supabase/ui';
+import { usePreFetchData } from 'context/PreFetchData';
+import {
+    JournalClient,
+    JournalSelector,
+    parseJournalDocuments,
+} from 'data-plane-gateway';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { useLocalStorage } from 'react-use';
+import getGatewayAuthConfig from 'services/gateway-auth-config';
+import useSWR from 'swr';
+import {
+    getStoredGatewayAuthConfig,
+    LocalStorageKeys,
+} from 'utils/localStorage-utils';
+
+enum ErrorFlags {
+    TOKEN_NOT_FOUND = 'Unauthenticated',
+    TOKEN_INVALID = 'Authentication failed',
+    OPERATION_INVALID = 'Unauthorized',
+}
+
+const useJournalsForCollection = (collectionName: string) => {
+    const { session } = Auth.useUser();
+    const { grantDetails } = usePreFetchData();
+
+    const [gatewayConfig, setGatewayConfig] = useLocalStorage(
```

Comment:
I'm not sure whether local storage is appropriate for this. The JWT that's used for accessing the data-plane may be different for each collection, so it seems potentially incorrect to use the same local storage key for all of them. Additionally, they may have relatively short expirations, and are fairly cheap to generate, so I don't see a reason to persist them indefinitely.

Reply:
This was roughly lifted from: 

https://github.com/estuary/ui/blob/0f4e1645a6a11de756722c9f99c523cf0665d212/src/hooks/useShardsList.ts#L27-L30

Should that not use localstorage either? Or is the token generation different in the case of reading journal data vs shard info?

Reply:
Oh this is news to me (I think) so I am going to setup a meeting with all of us to discuss.

Reply:
Had a quick meeting and decided that we will no longer store the token in LS and just keep it in memory. The UI will attempt to refresh the token at the proper interval. If someone reloads the page then we will fetch a fresh token.